specs/protocol/deputy-pause-module.md

```diff
@@ -0,0 +1,387 @@
+
+- Address of the Operations Safe.
+- Address of the `DeputyGuardianModule`.
+- Address of the `SuperchainConfig` contract.
+- Address of the Deputy account.
+
+#### Mitigations
+
+- Verify a signature from the Deputy in the contract constructor.
+- Verify a known getter function for each provided contract.
+- Verify the configured values in the deployment script.
+- Generate and test a signature on testnet.
+
+### aDPM-002: Deputy key is not compromised
+
+We assume that the Deputy key is maintained securely and is not compromised by a malicious actor.
+
+#### Mitigations
+
+- Enforce strong access control around the Deputy key.
+- Monitoring for any usage of the Deputy key.
+
+### aDPM-003: Deputy key is not deleted
+
+We assume that the Deputy key has not been deleted by a malicious actor.
+
+#### Mitigations
+
+- Maintain duplicate copies of the key in multiple secure locations.
+- Regularly generate valid signatures from the key.
+
+### aDPM-004: Deputy key will not sign excessively long expirations
+
+We assume that the Deputy key will not create authorization signatures that include expiration
+timestamps that are far into the future. The protocol for the honest Deputy is to only create
+signatures with a maximum lifetime that matches the maximum lifetime configured in the contract.
+
+#### Mitigations
+
+- Careful testing for software components that interact with the key.
+- Strict protocols for humans that interact with the key.
+
+### aDPM-005: Ethereum will not censor transactions for extended periods of time
+
+We assume that Ethereum will not censor transactions for an extended period of time and that any
+transaction submitted to the network can be processed within a reasonable time bound (e.g., 1h).
+
+#### Mitigations
+
+N/A
+
+### aDPM-006: DeputyGuardianModule pause function correctly triggers pause
+
+We assume that the `pause` function exposed by the `DeputyGuardianModule` will correctly trigger
+the Superchain-wide pause through the Guardian account. We assume that this function will not carry
+out any other action.
+
+#### Mitigations
+
+- Existing audit on the `DeputyGuardianModule`.
+
+### aDPM-007: OpenZeppelin ECDSA and EIP712 contracts are free of critical bugs
+
+We assume that both the `ECDSA` library and the `EIP712` contract provided by OpenZeppelin V4 at
+commit [ecd2ca2c](https://github.com/OpenZeppelin/openzeppelin-contracts/tree/ecd2ca2c) (v4.7.3)
+are free of any critical bugs that would cause their behaviors to diverge from their specified
+behaviors.
+
+#### Mitigations
+
+- Existing audits/safety processes for OpenZeppelin V4.
+
+### aDPM-008: Safe contracts are free of critical bugs
+
+We assume that the Safe contract implementations used by the Security Council Safe and Optimism
+Foundation Operations Safe are free of any critical bugs that would cause their behaviors to
+diverge from their specified behaviors.
+
+#### Mitigations
+
+- Existing audits/safety processes for Safe.
+
+## Invariants
```

Comment:
The impact/mitigations content in this doc is very similar to what we put into an FMA. I appreciate the thoroughness, but I'm worried we're duplicating effort. I think it would be sufficient to put invariants into a bulleted tree with dependencies as children.

Reply:
As much as possible I want to shift-left by having all of these impacts listed as invariants. This way we can understand the security model *before* we write all of the FMA stuff. IMO the FMA is a weak replacement for invariants in this case.